core/src/main/scala/org/apache/spark/internal/io/HadoopMapReduceCommitProtocol.scala

```diff
@@ -127,14 +127,13 @@ class HadoopMapReduceCommitProtocol(
     val filename = getFilename(taskContext, spec)
 
     val stagingDir: Path = committer match {
-      // For FileOutputCommitter it has its own staging path called "work path".
       case f: FileOutputCommitter =>
         if (dynamicPartitionOverwrite) {
           assert(dir.isDefined,
             "The dataset to be written must be partitioned when dynamicPartitionOverwrite is true.")
           partitionPaths += dir.get
         }
-        new Path(Option(f.getWorkPath).map(_.toString).getOrElse(path))
+        f.getTaskAttemptPath(taskContext)
```

Comment:
The work path field is set by a variant of `getTaskAttemptPath` method, which is private for overriding. Changes here make sure we look up the task attempt path consistently across setup -> commit/abort.